Engine/gapi/vulkan/vcommandbuffer.cpp

```diff
@@ -459,6 +459,17 @@ void VCommandBuffer::dispatch(size_t x, size_t y, size_t z) {
   vkCmdDispatch(impl,uint32_t(x),uint32_t(y),uint32_t(z));
   }
 
+void VCommandBuffer::dispatchIndirect(const AbstractGraphicsApi::Buffer& indirect, size_t offset) {
+  const VBuffer& ind = reinterpret_cast<const VBuffer&>(indirect);
+
+  curUniforms->ssboBarriers(resState, PipelineStage::S_Compute);
+  // block future writers
+  resState.onUavUsage(ind.nonUniqId, NonUniqResId::I_None, PipelineStage::S_Indirect);
+  resState.flush(*this);
```

Comment:
What barrier will be emitted, if buffer is in used by both shader and indirect argument?
Should be `SRC -> CS|indirect`, plus buffer has to be in read-state.

Reply:
Created test that simulates this situation:
barrier {UavReadComp | UavWriteComp -> UavReadComp | UavWriteComp | Indirect}

Propably stupid question :) but does read-state here refer to the condition that previous memory accesses for the buffer are available and visible for reading (i.e. that required barrier exists)? My first thought was about some internal state in the engine but couldn't find it.

Reply:
I mean `ResourceState::uavRead`. Need to make sure that subsequent command are aware that indirect-buffer is in use.
What I'm worry about, that `resState.flush` might reset tracking preemptively.

Basically in scenario 
```
dispath( write to buf )
--- barrier {UavReadComp | UavWriteComp -> UavReadComp | UavWriteComp | Indirect}
dispathIndirect(buf)
--- should be execution barrier indirect -> CS
dispath( write to buf )
```